Tests/Services/PactContractTests2.swift

```diff
@@ -0,0 +1,155 @@
+    try builder
+      .uponReceiving("a request for an object")
+      .given("some state")
+      .withRequest(method: .GET, path: "/object")
+      .willRespond(with: 200) { response in
+        try response.jsonBody(
+          .eachLike([
+            "id": .integer(1),
+            "name": .like("Test Object")
+          ])
+        )
+      }
+    
+    try await builder.verify { context in
+      let url = URL(string: "\(context.mockServerURL)/object")!
+      _ = try await self.session.data(from: url)
+    }
+    
+    try pact.writePactFile()
+    
+    let firstInteraction = try XCTUnwrap(getInteractions().first as? [String: Any])
+    
+    let response = try XCTUnwrap(firstInteraction["response"] as? [String: Any])
+
+    assertInlineSnapshot(of: response, as: .json) {
+      #"""
+      {
+        "body" : {
+          "content" : [
+            {
+              "id" : 1,
+              "name" : "Test Object"
+            }
+          ],
+          "contentType" : "application\/json",
+          "encoded" : false
+        },
+        "headers" : {
+          "Content-Type" : [
+            "application\/json"
+          ]
+        },
+        "matchingRules" : {
+          "body" : {
+            "$" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "type"
+                }
+              ]
+            },
+            "$[*].id" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "integer"
+                }
+              ]
+            },
+            "$[*].name" : {
+              "combine" : "AND",
+              "matchers" : [
+                {
+                  "match" : "type"
+                }
+              ]
+            }
+          }
+        },
+        "status" : 200
+      }
+      """#
+    }
+  }
+  
+  func test_object2() async throws {
+    try builder
+      .uponReceiving("a request for another object")
+      .given("some state")
+      .withRequest(method: .GET, path: "/object2")
+      .willRespond(with: 200) { response in
+        try response.jsonBody(
+          .eachLike([
+            "id": .integer(1),
+            "name": .like("Test Object")
+          ])
+        )
+      }
+    
+    try await builder.verify { context in
+      let url = URL(string: "\(context.mockServerURL)/object2")!
+      _ = try await self.session.data(from: url)
+    }
+    
+    try pact.writePactFile()
+    
+    let interactions = try getInteractions()
+    
+    XCTAssertEqual(1, interactions.count)
+  }
+}
+
+
+extension PactContractTests2 {
+  func getJsonObject() throws -> [String: Any] {
+    let fileContents = try String(contentsOfFile: "/tmp/pacts/sanity-consumer-sanity-provider.json")
+    guard let data = fileContents.data(using: .utf8),
+          let jsonObject = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] else {
+      return [:]
+    }
```

Comment:
thought: This is obviously a very subjective thing. Though if there's guarding against multiple optionals, maybe moving them all to their own lines could be easier to read? If there's only one, then `guard expression else {` is fine on one line. IMO, and again, this is very subjective, and it ties into the comment I left above regarding sticking to code style across the project and project timeline.

```swift
guard 
    let data = fileContents.data(using: .utf8),
    let jsonObject = try JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] 
else {
    return [:]
}
```

note-to-self: Should add/update `guard let` code style in [PactSwift ADRs](https://github.com/surpher/PactSwift/tree/main/Documentation/ADRs).